jcommune-model/src/test/java/org/jtalks/jcommune/model/entity/TopicTest.java

```diff
@@ -69,31 +69,25 @@ public void addPostShouldUpdateModificationDate() throws InterruptedException {
         Topic topic = createTopic();
         DateTime prevDate = topic.getModificationDate();
         Thread.sleep(25); // millisecond precise is a kind of fiction
-        topic.addPost(new Post());
+        Post post = new Post();
+        post.setCreationDate(new DateTime());
+        topic.addPost(post);
 
         assertTrue(topic.getModificationDate().isAfter(prevDate));
     }
 
-
-    public void updatePostShouldUpdateModificationDate() throws InterruptedException {
-        Topic topic = createTopic();
-        DateTime prevDate = topic.getModificationDate();
-        Thread.sleep(25); // millisecond precise is a kind of fiction
-        topic.getPosts().get(0).updateModificationDate();
-
-        assertTrue(topic.getModificationDate().isBefore(prevDate));
-    }
-
     @Test
-    public void updateModificationDateShouldChangeTheModificationDate() {
+    public void addPostShouldSetModificationDateToPostCreationDate() throws InterruptedException {
         Topic topic = createTopic();
         DateTime prevDate = topic.getModificationDate();
-
-        DateTime modDate = topic.updateModificationDate();
-
-        assertNotSame(modDate, prevDate);
+        Thread.sleep(25);
```

Comment:
I think, `Thread.sleep()` not a best choise in this case. You can set date manually and use methods like `date.plus Days()` to change it.